### PR TITLE
XTypes: fixes to wireshark for xtypes

### PR DESCRIPTION
**Pull request by ClaytonCalabrese**
small function call changes to deal with updates that are part of xtypes

### CODE REVIEW

File: tools/dissector/packet-opendds.cpp [hunk 1]
```diff
@@ -290,8 +290,9 @@ namespace OpenDDS
       offset += len;
 
       // hf_sequence
-      size_t size = 0, padding = 0;
-      gen_find_size(header.sequence_, size, padding);
+      size_t size = 0;
+      const Encoding encoding(Encoding::KIND_CDR_UNALIGNED);
+      serialized_size(encoding, size, header.sequence_);
```
Comment (iguessthislldo) [inline tools/dissector/packet-opendds.cpp:295]:
```suggestion
      size_t size = serialized_size(encoding, header.sequence_);
```

You can use this when the size starts at zero.

File: tools/dissector/packet-opendds.cpp [hunk 2]
```diff
@@ -347,8 +348,9 @@ namespace OpenDDS
       offset += len;
 
       // hf_sample_sequence
-      size_t size = 0, padding = 0;
-      gen_find_size(sample.sequence_, size, padding);
+      size_t size = 0;
+      const Encoding encoding(Encoding::KIND_CDR_UNALIGNED);
+      serialized_size(encoding, size, sample.sequence_);
```
Comment (iguessthislldo) [inline tools/dissector/packet-opendds.cpp:353]:
```suggestion
      size_t size = serialized_size(encoding, sample.sequence_);
```

File: tools/dissector/packet-opendds.cpp [hunk 3]
```diff
@@ -451,7 +453,7 @@ namespace OpenDDS
 
                   // Get Entry Size
                   size = 0;
-                  gen_find_size(filter, size, padding);
+                  serialized_size(encoding, size, filter);
```
Comment (iguessthislldo) [inline tools/dissector/packet-opendds.cpp:456]:
```suggestion
                  size = serialized_size(encoding, filter);
```

File: tools/dissector/packet-opendds.cpp [hunk 4]
```diff
@@ -290,8 +290,9 @@ namespace OpenDDS
       offset += len;
 
       // hf_sequence
-      size_t size = 0, padding = 0;
-      gen_find_size(header.sequence_, size, padding);
+      size_t size = 0;
+      const Encoding encoding(Encoding::KIND_CDR_UNALIGNED);
```
Comment (iguessthislldo) [inline tools/dissector/packet-opendds.cpp:294]: Could you factor the encoding objects out into a common one?

Comment (ClaytonCalabrese) [reply tools/dissector/packet-opendds.cpp:294]: Depends.  There are multiple functions.  I thought I did factor out to the minimum but I can take another look.

File: tools/dissector/packet-opendds.cpp (hunk 3 above)
Comment (ClaytonCalabrese) [reply tools/dissector/packet-opendds.cpp:456]: If I don't reset size back to 0 will it not cause problems?

File: tools/dissector/packet-opendds.cpp (hunk 4 above)
Comment (ClaytonCalabrese) [reply tools/dissector/packet-opendds.cpp:294]: Or are you saying to try and make it a single variable they all have access to within the namespace?

File: tools/dissector/packet-opendds.cpp (hunk 3 above)
Comment (iguessthislldo) [reply tools/dissector/packet-opendds.cpp:456]: This is a template wrapper around the generated `serialized_size` that passes a zero to it and returns the result. It's equivalent, just a bit smaller.

File: tools/dissector/packet-opendds.cpp (hunk 4 above)
Comment (iguessthislldo) [reply tools/dissector/packet-opendds.cpp:294]: > Or are you saying to try and make it a single variable they all have access to within the namespace?

Yes, I think you can just define a single encoding object for these cases and put it in the anonymous namespace higher up in the file.

**Comment by ClaytonCalabrese** [comment]
> Also were any changes needed for the sample dissector?

There are some of the old serializer constructor in data members that I will need to factor out in my other PR.  I'm not sure we can be certain until they are both together.

**Comment by ClaytonCalabrese** [comment]
> > Also were any changes needed for the sample dissector?
> 
> There are some of the old serializer constructor in data members that I will need to factor out in my other PR. I'm not sure we can be certain until they are both together.

After merging there are changes needed to the serializer endianess but nothing for this PR. I think this PR is ready to go.

**Pull request merged**